odoo/addons/base/security/base_security.xml

```diff
@@ -16,7 +16,7 @@
             the multi-company rule because it might interfere with the user's company rule
             and make some users unselectable in relational fields. This means that partners
             of internal users are always visible, not matter the company setting. -->
-            <field name="domain_force">['|', ('partner_share', '=', False), ('company_id', 'in', company_ids + [False])]</field>
+            <field name="domain_force">['|', '|', ('partner_share', '=', False), ('company_id', 'parent_of', company_ids), ('company_id', '=', False)]</field>
```

Comment:
You also need to change the `_check_company_domain`, it should always be synced (don't ask me why, I didn't chose it)